File: training-rest/src/main/java/com/julioluis/trainingrest/services/TrainingService.java
package com.julioluis.trainingrest.services;

import com.julioluis.trainingrest.entities.Training;

import java.util.List;

public interface TrainingService {

    List<Training> findAll();
    Training findById(Integer id);
    Training save(Training training);
    Training update(Training training);
    Training delete(Integer id);

}

File: training/src/api/services/TrainingService.js
import axios from 'axios'
import * as Constant from '../../Constant.js'

const URI='trainings/'

class TrainingService {

    fingAll=()=>{
        return axios.get(Constant.URL_BASE+URI);
    }

    findById=(trainingId)=> {
        return axios.get(Constant.URL_BASE+URI+trainingId);
    }

    save=(training)=> {
        return axios.post(Constant.URL_BASE+URI,training);
    }

    update=(training)=> {
        return axios.put(Constant.URL_BASE+URI,training);
    }

    delete=(trainingId)=> {
        return axios.delete(Constant.URL_BASE+URI+trainingId);
    }




}

export default new TrainingService();